server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -122,56 +120,61 @@ public CompletionStage<Void> restore(ZipFile zip) {
       ConfigurationManager configurationManager = SecurityActions.getGlobalComponentRegistry(cm).getComponent(ConfigurationManager.class);
       Properties properties = new Properties();
       properties.put(CacheParser.IGNORE_DUPLICATES, true);
-      for (String cacheName : resources) {
-         CompletionStage<Void> cs = blockingManager
-               .supplyBlocking(() -> recoverCache(cacheName, properties, configurationManager, zip), "restore-cache-" + cacheName)
-               .thenCompose(Function.identity());
-         stages.dependsOn(cs);
-      }
-      return stages.freeze();
+      return CompletionStages.performConcurrently(resources, ProcessorInfo.availableProcessors(), blockingManager.asExecutor("cache-restore"), cacheName -> {
```

Comment:
We should be using the non blocking executor here. No reason to use the blocking manager, it will actually make it worse due to how BlockingManager#blockingPublisher works.